src/devices/vdml_usd.cpp

```diff
@@ -21,5 +21,80 @@ std::int32_t is_installed(void) {
 	return usd_is_installed();
 }
 
+std::int32_t list_files_raw(const char* path, char* buffer, int32_t len) {
+	return usd_list_files_raw(path, buffer, len);
+}
+
+std::vector<std::string> list_files(const char* path, char* buffer, int32_t len) {
```

Comment:
This function should definitely not require users to pass in their own `buffer` and `len`. You should just allocate a buffer yourself at the top of this function body then free it after it's done being used.

Reply:
Implemented this change in commit 4247fb197e02a639c26dd75961c81d81e3e5b30a

Reply:
Looks good, can I ask if the number 10000 came from somewhere or was it chosen arbitrarily?

Reply:
Came from an estimate of the worst case that is realistic to happen. We needed to chose a number, and decided that 100 files in a single directory with an approximate average length of 100 (really 99 if you include the separators) will be enough for every feasible use case we could come up with.

Reply:
Ok, fair enough. If you're choosing a number, a power of 2 (or sum of powers of 2) might be nice for memory allocator friendliness :)

(I have no idea what allocation strategy VexOS uses, this may be completely irrelevant)